dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkAspect.java

```diff
@@ -80,7 +77,16 @@ public void onError() {
         try {
             rollbackTransaction();
         } finally {
-            session.close();
+            onFinish();
+        }
+    }
+
+    public void onFinish() {
```

Comment:
According to the docs, it sounds like this is called on success or failure: 

https://github.com/jersey/jersey/blob/5f9e5fb78d48ce45431032a5867c157f191716fc/core-server/src/main/java/org/glassfish/jersey/server/monitoring/RequestEvent.java#L174-L179